src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/GerritServer.java

```diff
@@ -914,14 +915,21 @@ private void rename(String newName) {
      * @param oldName the old name of the Gerrit server
      */
     private void changeSelectedServerInJobs(String oldName) {
-        for (AbstractProject job : PluginImpl.getConfiguredJobs_(oldName)) {
-            GerritTrigger trigger = (GerritTrigger)job.getTrigger(GerritTrigger.class);
+        for (Job job : PluginImpl.getConfiguredJobs_(oldName)) {
+
+            // TODO: find a way to handle add/remove of triggers
+            if (!(job instanceof AbstractProject)) {
+                return;
```

Comment:
On Jul 27, 2015 12:48 AM, "Tom Fennelly" <notifications@github.com> wrote:
>
> In
src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/GerritServer.java:
>
> > @@ -914,14 +915,21 @@ private void rename(String newName) {
> >       * @param oldName the old name of the Gerrit server
> >       */
> >      private void changeSelectedServerInJobs(String oldName) {
> > -        for (AbstractProject job :
PluginImpl.getConfiguredJobs_(oldName)) {
> > -            GerritTrigger trigger =
(GerritTrigger)job.getTrigger(GerritTrigger.class);
> > +        for (Job job : PluginImpl.getConfiguredJobs_(oldName)) {
> > +
> > +            // TODO: find a way to handle add/remove of triggers
> > +            if (!(job instanceof AbstractProject)) {
> > +                return;
>
> Auto removing zombie triggers after a Gerrit server is removed seems to
make sense to me.
>
> In any case (and imo), the debate on the pros and cons of
programmatically removing a trigger is something that should happen
separate to this PR. IMO, our job here is to integrate workflow and
maintain the same feature set. It shouldn't be about proposing fundamental
plugin design changes.
>
> —
> Reply to this email directly or view it on GitHub.

While that is true.. I don't know of a good way to implement this
functionality and it may make sense to do the PR to remove such a feature
first unless we can actually do this change..

One way might be to optionally depend on work flow plug in.  I have seen
others do this but I don't know how they did it.